src/anomalib/post_processing/post_processor.py

```diff
@@ -135,15 +135,15 @@ def on_validation_epoch_end(self, trainer: Trainer, pl_module: LightningModule)
         del trainer, pl_module
         if self.enable_thresholding:
             # compute threshold values
-            if self._image_threshold_metric.update_called:
+            if getattr(self._image_threshold_metric, "_update_called", False):
```

Comment:
I am personally not in favour of using `getattr`s if possible. If the image threshold metric isn't set, it will be None. In that case we can just compare it against null value.